src/components/Filters.js

```diff
@@ -1,39 +1,56 @@
-import { useEffect } from "react";
-import { BsChevronDown } from "react-icons/bs";
 import { MdOutlineKeyboardAlt } from "react-icons/md";
 
 
-const Filters = ({ tag, handleTag, data, tagsArray, filter }) => {
-  useEffect(() => {
-    if (tag !== "") {
-    }
-
-  }, [tag])
+const Filters = ({ tag, handleTag, resetData, tagsArray, filter, handleMidTag, midTag, handleFilterInput, filterInput }) => {
+  // console.log(tag)
+  
   const handleCategoryChange = (e) => {
+    resetData()
+    handleMidTag("")
     handleTag(e.target.value)
+    handleFilterInput("")
+  }
+
+
+  const handleMidTagChange = (e) => {
+    resetData()
+    handleMidTag(e.target.value)
+    handleFilterInput("")
   }
+  const handleFilterInputChange = (e) => {
+      resetData()
+      handleFilterInput(e.target.value)
+  }
+  const filterOptions = Object.keys(tagsArray[0]).length && Object.keys(tagsArray[0])
+
   return (
     <div className="filters flex-row">
       <select className="padding border" id="tags" defaultValue={tag} onChange={handleCategoryChange}>
```

Comment:
onChange = {handleTag}

in the parent component you have handleTag, there add resetFilters()